packaging/installer/netdata-updater.sh

```diff
@@ -87,32 +89,85 @@ _cannot_use_tmpdir() {
 }
 
 create_tmp_directory() {
-  if [ -z "${TMPDIR}" ] || _cannot_use_tmpdir "${TMPDIR}" ; then
-    if _cannot_use_tmpdir /tmp ; then
-      if _cannot_use_tmpdir "${PWD}" ; then
-        echo >&2
-        echo >&2 "Unable to find a usable temprorary directory. Please set \$TMPDIR to a path that is both writable and allows execution of files and try again."
-        exit 1
+  if [ -n "${NETDATA_TMPDIR_PATH}" ]; then
+    echo "${NETDATA_TMPDIR_PATH}"
+  else
+    if [ -z "${TMPDIR}" ] || _cannot_use_tmpdir "${TMPDIR}" ; then
+      if _cannot_use_tmpdir /tmp ; then
+        if _cannot_use_tmpdir "${PWD}" ; then
+          echo >&2
+          echo >&2 "Unable to find a usable temprorary directory. Please set \$TMPDIR to a path that is both writable and allows execution of files and try again."
+          exit 1
+        else
+          TMPDIR="${PWD}"
+        fi
       else
-        TMPDIR="${PWD}"
+        TMPDIR="/tmp"
       fi
-    else
-      TMPDIR="/tmp"
     fi
-  fi
 
-  mktemp -d -t netdata-updater-XXXXXXXXXX
+    mktemp -d -t netdata-updater-XXXXXXXXXX
+  fi
 }
 
-download() {
+_safe_download() {
   url="${1}"
   dest="${2}"
   if command -v curl > /dev/null 2>&1; then
-    curl -sSL --connect-timeout 10 --retry 3 "${url}" > "${dest}" || fatal "Cannot download ${url}"
+    curl -sSL --connect-timeout 10 --retry 3 "${url}" > "${dest}"
+    return $?
   elif command -v wget > /dev/null 2>&1; then
-    wget -T 15 -O - "${url}" > "${dest}" || fatal "Cannot download ${url}"
+    wget -T 15 -O - "${url}" > "${dest}"
+    return $?
   else
+    return 255
+  fi
+}
+
+download() {
+  url="${1}"
+  dest="${2}"
+
+  _safe_download "${url}" "${dest}"
+  ret=$?
+
+  if [ ${ret} -eq 0 ]; then
+    return 0
+  elif [ ${ret} -eq 255 ]; then
     fatal "I need curl or wget to proceed, but neither is available on this system."
+  else
+    fatal "Cannot download ${url}"
+  fi
+}
+
+newer_commit_date() {
+  echo >&3 "Checking if a newer version of the updater script is available."
+
+  if command -v jq > /dev/null 2>&1; then
+    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | jq '.[0].commit.committer.date')"
+  elif command -v python > /dev/null 2>&1;then
+    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | python -c 'from __future__ import print_function;import sys,json;print(json.load(sys.stdin)["commit"]["committer"]["date"])')"
```

Comment:
I guess, trying to install netdata and run the updater on a system that doesn't have support for either of these (jq or python) is an edge case and not possible.

Reply:
No, it will work fine. We set `commit_date` to the start of the UNIX epoch below if it’s unset, so not having either of these conditions match just means that systems that have neither `jq` nor Python will always download the updater script instead of only doing it when there’s a newer version in the repo.